### DISCUSSION TITLE
Newbie stumped: Why are apps not launching?

### DISCUSSION THREAD

**Discussion by rbanks88** [ttscoff/bunch-docs#184] [Q&A] [answered]
There may be other problems with this bunch, but for starters the two apps I want to launch aren't launching. You just put their names at the start of a line, right? 

The Console just reports:
2022-01-12 13:01:31: 🔴App not running (BusyCal)
2022-01-12 13:01:31: 🔴App not running (OmniFocus)

Here's my script:

```
---
title: Morning Startup
---

// Launch BusyCal in week view with today in the middle
BusyCal
- {@0 !1} // Show BusyCal window via Cmd 0 with 1 sec. pause after
- {@2} // Set weekly view
- {@t} // Go to today
- {left/←} // Left arrow, show prior day at left
- {left/←} // Left arrow, show prior day at left
* "Hide BusyCal Sidebars.scpt" // runs applescript in /Bunches that runs Keyboard Maestro macro

OmniFocus
- {~@w} // Close all windows via Opt Cmd w
- {@1 !1} // Open inbox with 1 sec. pause after
* "Hide OmniFocus Sidebar.scpt" // runs applescript in /Bunches that runs Keyboard Maestro macro
* "Show OmniFocus Inspector.scpt" // runs applescript in /Bunches that runs Keyboard Maestro macro
- {~@n !1} //O pen new window via Opt Cmd n with 1 sec. pause after
* "Show OmniFocus Sidebar.scpt" // runs applescript in /Bunches that runs Keyboard Maestro macro
- {@1} // Show inbox





	<<#Position Inbox ~2 // Calls snippet below, with additional 2 sec. delay

___ // 3+ underscores mean items below will be read as a snippet file.
--[Position Inbox]
* tell application "Moom" to arrange windows according to snapshot "Clear Inbox"
```

Even this bunch, made from the "New Bunch with Open Apps" command, doesn't launch OmniFocus:

```
---
title: All open apps
---
OmniFocus

```

**Comment by ttscoff** [comment, marked as answer]
Did you grant permissions (accessibility) to Bunch?

**Reply by rbanks88** [reply]
Yes, I granted access. permission to Bunch. Interesting, though, mine says “Bunch Beta.app” and the illustration on the Bunch site is just “Bunch.app.” I downloaded it yesterday, v. 1.4.7.

Also, the Bunch install page says, “As Bunch attempts to control various apps and features of the operating system, you will be asked to grant Bunch permission to perform new operations. Because Bunch is all about automating other apps, your first runs will include a lot of these dialogs.”

I don’t recall seeing such dialogs, but if I had, I would have granted it permissions.

I'd restart the Mac (M1 MacBook Air), but I'm running a CC Cloner backup of my photos disk, which will be running another 18 hours...

**Reply by rbanks88** [reply]
OK. It started working when I unchecked Toggle Bunches in the prefs. However, when I ran a bunch that launched Mail, OmniFocus and BusyCal, Mail and OmniFocus showed on top of all my other running apps as they launched, but BusyCal's window didn't. Is it normal for apps to launch on top? I don't see a Bunch command to bring an app to the front, though I do see the one to hide an app.

**Comment by ttscoff** [comment]
Toggle Bunches definitely shouldn't affect the permissions issue, so it 
might be worth toggling that back on, attempting a reboot, and seeing if 
that's what actually fixed it.

Is BusyCal already running (possibly as a background/menu bar helper)? 
It won't come to the front if it's not actually launching. The way to 
force apps to focus when launching is to append a caret, e.g. 
`BusyCal^`.

-Brett
